apps/outlook-rooms-addin/src/app/rooms/features-filter.service.ts

```diff
@@ -33,43 +38,56 @@ export class FeaturesFilterService {
         { name: 'Jamboard', id: 'Jamboard', value: false },
         { name: 'Projector', id: 'Projector', value: false },
         { name: 'Views', id: 'Views', value: false },
-    ]);
+    ];
 
     constructor(private _state: EventFormService) {
-        this.features$ = this.Features;
-        this._getSelectedFeatures();
+        this._features.next(this.room_features);
     }
 
-    private _getSelectedFeatures() {
+    async getSelectedFeatures() {
         this.selected_features$ = this.features$.pipe(
             map((arr) => arr.filter((item) => item.value == true))
         );
+        await this.selected_features$.pipe(take(1)).toPromise();
+        this.selected_features$?.subscribe(this._selected_features);
     }
     async applyFilter() {
-        let selected_features_list = await this.selected_features$
-            .pipe(take(1))
-            .toPromise();
+        await this.selected_features$.pipe(take(1)).toPromise();
+
+        const requested_features = await this.sortSelectedFeatures(
+            this.selected_features
+        );
 
-        this.features_list = selected_features_list.map((item) => item.id);
+        this.updated_spaces$ = this.spaces$.pipe(
+            map((spaces: Space[]) =>
+                spaces.filter((space: Space) => {
+                    return this._sort(space.feature_list).includes(
+                        requested_features
+                    );
+                })
+            )
+        );
+        await this.updated_spaces$?.pipe(take(1)).toPromise();
+        this.updated_spaces_emitter.next(true);
+    }
 
-        let current_spaces = await this.spaces$.pipe(take(1)).toPromise();
+    _sort(array: string[]): string {
```

Comment:
Can you name this more appropriately as it this method does more than just a "sort".